lib/pure/math.nim

```diff
@@ -62,6 +62,8 @@ when defined(c) or defined(cpp):
   proc c_isnan(x: float): bool {.importc: "isnan", header: "<math.h>".}
     # a generic like `x: SomeFloat` might work too if this is implemented via a C macro.
 
+  proc c_copysign[T: SomeFloat](x, y: T): T {.importc: "copysign", header: "math.h".}
```

Comment:
generic importc doesn't make sense unless you know the C implementation uses a macro, which I don't think is the case, at least in many libc implementations

2 options:
* P1 (simplest): make  c_copysign not generic
but it still needs a VM fallback (maybe via vmops), ditto for js

* P2: add c_copysign, c_copysignl, c_copysignf
(ditto for vm etc)

* P3: or, use/adapt this implementation: $git_clone_D/zig/lib/libc/mingw/math/copysignf.c
```c
/**
 * This file has no copyright assigned and is placed in the Public Domain.
 * This file is part of the mingw-w64 runtime package.
 * No warranty is given; refer to the file DISCLAIMER.PD within this package.
 */
#include <math.h>

typedef union ui_f {
	float f;
	unsigned int ui;
} ui_f;

float copysignf(float aX, float aY)
{
  ui_f x,y;
  x.f=aX; y.f=aY;
  x.ui= (x.ui & 0x7fffffff) | (y.ui & 0x80000000);
  return x.f;
}
```

this could be turned into a generic to work with float32|float64, eg:
```nim
type FloatAsUint[T] {.union, packed.} = object
  a: T
  when T.sizeof == 32:
    f: uint32
  else:
    ui: uint64

proc copySign[T: SomeFloat](a, b: T): T =
  var x, y: FloatAsUint[T]
  x.f=a; y.f=b;
  x.ui = (x.ui and 0x7fffffff) or (y.ui & 0x80000000) # needs to be adapted for 32 vs 64
  result = x.f
}
```